docs/contributing/coding-guidelines/ros-nodes/class-design.md

```diff
@@ -1,5 +1,67 @@
 # Class design
 
+We'll use the `autoware_gnss_poser` package as an example.
+
+## Namespaces
+
+```cpp
+namespace autoware::gnss_poser
+{
+...
+} // namespace autoware::gnss_poser
+```
+
+- Everything should be under `autoware::gnss_poser` namespace.
+- Closing braces should contain a comment with the namespace name.
+
+## Classes
+
+### Nodes
+
+#### `gnss_poser.hpp`
```

Comment:
following the directory structure guideline, should this file be renamed `gnss_poser_node.hpp` ?

Reply:
Thanks, you are right, will fix it.

Reply:
Done with: https://github.com/autowarefoundation/autoware-documentation/pull/547/commits/c59e128a2dc1482f65e0acc274f34ff409da70c0